probe-rs/targets/STM32WB_Series.yaml

```diff
@@ -1,484 +1,810 @@
-            end: 0x8080000
-          is_boot_memory: true
-          cores:
-            - main
+        name: Main_Flash
+        range:
+          start: 0x8000000
+          end: 0x8080000
+        is_boot_memory: true
+        cores:
+          - main
     flash_algorithms:
-      - stm32wb_m4
+      - stm32wb5x_m4_512
+      - stm32wb55x_35x_m4_opt
   - name: STM32WB55VGQx
     cores:
       - name: main
         type: armv7em
-        core_access_options:
-          !Arm
-            ap: 0x0
-            psel: 0x0
+        core_access_options: !Arm
+          ap: 0
+          psel: 0
     memory_map:
       - !Ram
-          range:
-            start: 0x20000000
-            end: 0x20040000
-          is_boot_memory: false
-          cores:
-            - main
+        name: SRAM1
+        range:
+          start: 0x20000000
+          end: 0x20030000
+        cores:
+          - main
       - !Nvm
-          range:
-            start: 0x8000000
-            end: 0x8100000
-          is_boot_memory: true
-          cores:
-            - main
+        name: Main_Flash
+        range:
+          start: 0x8000000
+          end: 0x8100000
+        is_boot_memory: true
+        cores:
+          - main
     flash_algorithms:
-      - stm32wb_m4
+      - stm32wb5x_m4_1024
+      - stm32wb55x_35x_m4_opt
   - name: STM32WB55VGYx
-    part: 0x495
     cores:
       - name: main
         type: armv7em
-        core_access_options:
-          !Arm
-            ap: 0x0
-            psel: 0x0
+        core_access_options: !Arm
+          ap: 0
+          psel: 0
+    memory_map:
+      - !Ram
+        name: SRAM1
+        range:
+          start: 0x20000000
+          end: 0x20030000
+        cores:
+          - main
+      - !Nvm
+        name: Main_Flash
+        range:
+          start: 0x8000000
+          end: 0x8100000
+        is_boot_memory: true
+        cores:
+          - main
+    flash_algorithms:
+      - stm32wb5x_m4_1024
+      - stm32wb55x_35x_m4_opt
+  - name: STM32WB55VYYx
+    cores:
+      - name: main
+        type: armv7em
+        core_access_options: !Arm
+          ap: 0
+          psel: 0
+    memory_map:
+      - !Ram
+        name: SRAM1
+        range:
+          start: 0x20000000
+          end: 0x20030000
+        cores:
+          - main
+      - !Nvm
+        name: Main_Flash
+        range:
+          start: 0x8000000
+          end: 0x80a0000
+        is_boot_memory: true
+        cores:
+          - main
+    flash_algorithms:
+      - stm32wb5x_m4_640
+      - stm32wb55x_35x_m4_opt
+  - name: STM32WB5MMGHx
+    cores:
+      - name: main
+        type: armv7em
+        core_access_options: !Arm
+          ap: 0
+          psel: 0
     memory_map:
       - !Ram
-          range:
-            start: 0x20000000
-            end: 0x20040000
-          is_boot_memory: false
-          cores:
-            - main
+        name: SRAM1
+        range:
+          start: 0x20000000
+          end: 0x20040000
```

Comment:
I think this should be `0x20030000`. The module's datasheet says it's a STM32WB55VGY and that has 192K SRAM1 (256K total but SRAM2 is generally missing from this file.)

Reply:
The B55 series is i think the beefier version, I think. The  `STM32WB1M` is based off of the `STM32WB15CCY` which has 320/48 kbyte total.

Reply:
> The STM32WB1M is based off

But this comment is on the `WB5MMG` and according to the `WB5MMG` datasheet:

> Based on the STMicroelectronics STM32WB55VGY wireless microcontroller

I think the `STM32WB1M` looked fine when I checked that datasheet.

Reply:
Sorry, to finish my point, the 55V should have 192K RAM (0x30000) according to:
![image](https://github.com/probe-rs/probe-rs/assets/977627/2efb5f09-87bd-4127-9a8a-7ba0637829be)

Reply:
Ah, I see now.  

Datasheet has SRAM1 as 192Kbyte @ `0x2000_0000` so `0x2003_0000` does make a lot more sense.


I think sram2 missing is intentional; the ST radio seems to have exclusive access to that (and the second CPU core...) by default on the WB family)

Reply:
Addressed here: https://github.com/kquinsland/probe-rs/commit/952e3d3abaf6ca830a396fd09695e791a9b0a7b8